.github/workflows/main.yml

```diff
@@ -43,17 +43,27 @@ jobs:
         with:
           failure-threshold: error
 
-  docker-publish:
-    if: github.ref == 'refs/heads/main'
+  tag-new-version:
+    # This uses `conventional commits` to generate tags.  A full list
+    # of valid prefixes is here:
+    # https://github.com/commitizen/conventional-commit-types/blob/master/index.json)
+    #
+    # fix, perf -> patch release
+    # feat -> minor release
+    # BREAKING CHANGE in footer -> major release
+    #
+    # anything else (docs, refactor, etc) does not create a release
     runs-on: ubuntu-latest
-    needs: [test, lint-dockerfile]
-    env:
-      image: ghcr.io/opensafely-core/cohortextractor-v2:latest
+    outputs:
+      tag: ${{ steps.tag.outputs.new_version }}
     steps:
       - uses: actions/checkout@v2
-      - name: Build image
-        run: docker build . --file Dockerfile --tag ${{ env.image }}
-      - name: Log into GitHub Container Registry
-        run: docker login https://ghcr.io -u ${{ github.actor }} --password ${{ secrets.DOCKER_RW_TOKEN }}
-      - name: Push image to GitHub Container Registry
-        run: docker push ${{ env.image }}
+        with:
+          fetch-depth: 0
+      - name: Bump version and push tag
+        id: tag
+        uses: mathieudutour/github-tag-action@981ffb2cc3f2b684b2bfd8ee17bc8d781368ba60
```

Comment:
I'm considering using `github-tag-action` for reusable actions. Why use this commit, specifically? And not, for example, the corresponding tag? (v4.5)

Reply:
Tags are mutable so this ensures we're pointing at the expected commit should they update v4.5 to point at something else.

Reply:
3rd party actions should be pinned to sha, not tag, so an attacker can't change the tag: https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#using-third-party-actions

1st part actions do not have that restriction, as we control the provenance of the tag.

Reply:
Thanks for the comment and the  link, @bloodearnest.